packages/mds-policy-service/service/provider.ts

```diff
@@ -108,6 +108,13 @@ export const PolicyServiceProvider: ServiceProvider<PolicyService, PolicyService
         policy_id,
         policy_metadata: { intent_type: intent_draft.intent_type }
       })
-      return PolicyRepository.publishPolicy(policy_id)
+      const published_date = now()
+      return PolicyRepository.publishPolicy(policy_id, published_date, {
+        beforeCommit: process.env.KAFKA_HOST
+          ? async policy => {
+              return await PolicyStreamKafka.write({ ...policy, published_date })
+            }
```

Comment:
nit, unnecessary to await a returned promise
```suggestion
          ? async policy => PolicyStreamKafka.write({ ...policy, published_date })
```